runners/gradle-plugin/src/main/kotlin/org/jetbrains/dokka/gradle/dokkaConfigurations.kt

```diff
@@ -0,0 +1,41 @@
+package org.jetbrains.dokka.gradle
+
+import org.gradle.api.Project
+import org.gradle.api.artifacts.Configuration
+import org.gradle.api.attributes.Usage
+
+internal fun Project.maybeCreateDokkaDefaultPluginConfiguration(): Configuration {
```

Comment:
Can we `maybeRegister` them instead or is there a specific reason for creating them?

Reply:
Unfortunately, I think there is no such thing as "maybeRegister", but I could just register them in the `main.kt`. Since the tasks are configured lazy, the maybeCreate is fine for them! 

Let me see what we can do there! ☺️

Reply:
Unfortunately, there is no such thing as "maybeRegister" in Gradle. 
Looking at the configuration code, I think there is no reason why this should be evaluated lazy:

```kotlin
        extendsFrom(maybeCreateDokkaDefaultRuntimeConfiguration())
        attributes.attribute(Usage.USAGE_ATTRIBUTE, project.objects.named(Usage::class.java, "java-runtime"))
        isCanBeConsumed = false
        defaultDependencies { dependencies ->
            dependencies.add(project.dokkaArtifacts.dokkaCore)
        }
```

I could use the register function in main.kt, but do you think we really gain something here?